### PR TITLE
Pr/api window adapter

### PR DESCRIPTION
**Pull request by hunger**
Replace `PlatformWindow` with `WindowAdapter`.

Replace `create_window` with `create_window_adapter`.

Replace remaining `Platform Abstraction` to plain `Platform`.

Fixes: #1536 (what remained of that after the first PR).

### CODE REVIEW

File: tools/viewer/README.md [hunk 1]
```diff
@@ -64,25 +64,25 @@ slint-viewer --on open-url 'xdg-open $1' myfile.slint
...
 The program returns with the following error code:
- - If the command line argument parsing fails, the exit code will be *1*
- - If the .slint compilation fails, the compilation error will be printed to stderr and the exit code
-   will be *-1*
- - If a Window is closed, the exit code will be *0*
- - If a Dialog is closed with the "Ok" or "Closed" or "Yes" button, the exit code will be *0*
- - If a Dialog is closed with the "Cancel" or "No" button, or using the close button in the window
-   title bar, the exit code will be *1*
+
+- If the command line argument parsing fails, the exit code will be _1_
```
Comment (ogoffart) [inline tools/viewer/README.md:79]: Please reconfigure your reformatting script to ignore on markdown files (`*.md`)

**Review by ogoffart — CHANGES_REQUESTED** [review]
Please undo the reformating changes in README.md

File: internal/core/lib.rs [hunk 2]
```diff
@@ -85,11 +85,11 @@ pub type Coord = i32;
 /// Internal function to access the platform abstraction.
 /// The factory function is called if the platform abstraction is not yet
 /// initialized, and should be given by the platform_selector
-pub fn with_platform_abstraction<R>(
+pub fn with_platform(
```
Comment (ogoffart) [inline internal/core/lib.rs:88]: missing `<R>`   (as the CI points out)

File: internal/backends/winit/glwindow.rs [hunk 3]
```diff
@@ -17,13 +17,13 @@ use corelib::component::ComponentRc;
...
 use corelib::layout::Orientation;
-use corelib::window::{PlatformWindow, WindowHandleAccess};
+use corelib::window::{WindowAdapter, WindowHandleAccess};
 use corelib::Property;
 use corelib::{graphics::*, Coord};
 use i_slint_core as corelib;
 use winit::dpi::LogicalSize;
 
-/// GraphicsWindow is an implementation of the [PlatformWindow][`crate::eventloop::PlatformWindow`] trait. This is
+/// GraphicsWindow is an implementation of the [WindowAdapter][`crate::eventloopMyWindowAdapter`] trait. This is
```
Comment (ogoffart) [inline internal/backends/winit/glwindow.rs:26]: That looks wrong!

File: tools/viewer/README.md (hunk 1 above)
Comment (hunger) [reply tools/viewer/README.md:79]: Sorry for that: My upstream vim setup turned on format-on-save and I had not noticed it:-/

I have now forced that to off for everything.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**